### DISCUSSION TITLE
Unexpected behavior of `AUTH_REDIRECT_PROXY_URL`

### DISCUSSION THREAD

**Discussion by KacperAdamczyk** [nextauthjs/next-auth#11281] [Help]
Hello,
I would like to share my recent struggles with using the new redirect proxy feature and a confusing behavior that I've found during the process.
After reading the paragraph explaining the usage of this feature in the docs, it seemed pretty straightforward. I have to set the env variable on one of my environments pointing to the stable environment and ensure they share the same secret.
I was wrong. It took me a couple of days, countless tries, and digging through the source code to figure out that this env must be set on **both** environments, stable and preview, which wasn't clearly (at least for me) stated in the [docs](https://authjs.dev/getting-started/deployment#securing-a-preview-deployment). It only says about the preview deployment:

> In your preview deploy application, set `AUTH_REDIRECT_PROXY_URL` to [...]

If this is the intended behavior it would be great to improve this part of the docs so other people don't go through the same struggle I did😄 

I was using the latest beta version [next-auth@5.0.0-beta.19](https://github.com/nextauthjs/next-auth/releases/tag/next-auth%405.0.0-beta.19) and next@v14.2.4.

Thank you

**Comment by jvgusch** [comment]
Thank you for posting this! I have spent the last few weeks trying to figure out what I was doing wrong and was about to throw in the towel until I saw this discussion. Once I set `AUTH_REDIRECT_PROXY_URL` on the stable deployment as well as the preview it worked right away. This definitely feels like a bug or at least an error in the documentation that should be fixed.

**Comment by Zac-Zajdel** [comment]
@KacperAdamczyk THANK YOU SO MUCH.

I was beyond confused and felt so disappointed in myself for being unable to get this working.

I created a PR to the repo that hopefully will help mitigate this confusion for future devs - https://github.com/nextauthjs/next-auth/pull/12081

Onward and upward 🚀
